vulture_os/authentication/ldap/views.py

```diff
@@ -73,17 +74,24 @@ def ldap_view(request, object_id):
     return render(request, 'authentication/ldap_view.html', {"object_id": object_id})
 
 
-def ldap_edit(request, object_id=None):
+def ldap_edit(request, object_id=None, api=False):
     ldap = None
     if object_id:
         try:
             ldap = LDAPRepository.objects.get(pk=object_id)
         except ObjectDoesNotExist:
             return HttpResponseForbidden("Injection detected")
 
-    form = LDAPRepositoryForm(request.POST or None, instance=ldap, error_class=DivErrorList)
+    if hasattr(request, "JSON") and api:
+        form = LDAPRepositoryForm(request.JSON or None, instance=ldap, error_class=DivErrorList)
+    else:
+        form = LDAPRepositoryForm(request.POST or None, instance=ldap, error_class=DivErrorList)
 
     def render_form(objectid=None, **kwargs):
+        if api:
+            logger.error("Frontend api form error : {}".format(form.errors.get_json_data()))
+            return JsonResponse({"errors": build_form_errors(form.errors)}, status=400)
+
```

Comment:
This doesn't seem right...
Is that a forgotten copy/paste ?

Reply:
In case of an API call, the render form is called only if an error occured.